### PR TITLE
fix(gatsby-theme-bodiless): allow to pass props to gatsby img

### PR DESCRIPTION
**Pull request by beliayeu**
## Changes
* fixed the issue when image attrs including alt and title are not rendered when gatsby image is activated

### CODE REVIEW

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx [hunk 1]
```diff
@@ -24,24 +33,41 @@ type ImageProps = {
...
+const asGatsbyImage$ = (Component: CT<any>) => {
   const AsGatsbyImage = (props: GasbyImageProps) => {
     const { gatsbyImg, preset, ...rest } = props;
     if (gatsbyImg !== undefined) {
+      const gatsbyImgProps = pick(props, GATSBY_IMG_PROPS) as GatsbyImageOptionalProps;
       return (
-        <GatsbyImgWrapper {...rest}>
-          <GatsbyImg {...gatsbyImg} />
-        </GatsbyImgWrapper>
+        <GatsbyImg {...gatsbyImg} {...gatsbyImgProps} />
```
Comment (wodenx) [inline packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:70]: Do we want these props to override the presets?

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx [hunk 2]
```diff
@@ -24,24 +33,41 @@ type ImageProps = {
...
+];
 
-const asGatsbyImage = (Component: CT<any>) => {
+const isGatsbyImage = ({ gatsbyImg }: GasbyImageProps) => gatsbyImg !== undefined;
+
+const asGatsbyImage$ = (Component: CT<any>) => {
   const AsGatsbyImage = (props: GasbyImageProps) => {
     const { gatsbyImg, preset, ...rest } = props;
     if (gatsbyImg !== undefined) {
+      const gatsbyImgProps = pick(props, GATSBY_IMG_PROPS) as GatsbyImageOptionalProps;
```
Comment (wodenx) [inline packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:68]: Do we need to pick these props?  I'm concerned about maintainability here, and the worst case is a site builder will pass a bad prop and get a warning, no?

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx [hunk 3]
```diff
@@ -51,4 +77,10 @@ const asGatsbyImage = (Component: CT<any>) => {
   return AsGatsbyImage;
 };
 
+const asGatsbyImage = flow(
+  asGatsbyImage$,
+  ifToggledOn(isGatsbyImage)(withActivatorWrapper('onClick', Div)),
```
Comment (wodenx) [inline packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:82]: what's the use-case for when a gatsby image would not be a gatsby image?

Comment (wodenx) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:82]: Also - we only want this when editable, right?

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx (hunk 1 above)
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:70]: GatsbyImg component does not accept preset. We need pass image preset variations generated on the backend side. So preset will not have any effect here.

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx (hunk 2 above)
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:68]: agree. per quick gatsby image code check, it picks props itself, so it looks we are safe here
but will make a quick to validate this

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx (hunk 3 above)
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:82]: >>what's the use-case for when a gatsby image would not be a gatsby image?
the use case when image is uploaded but gatsby image variations are not generated yet
in this case we are displaying a standard bodiless image

Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx:82]: |Also - we only want this when editable, right?

right, will fix it

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx (hunk 2 above)
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx]: validated - no warnings are thrown, the invalid props are not rendered in the markup
removed the pick logic in ce9fb9b75a305e327cdf265fd377690f19c9191c

File: packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx (hunk 3 above)
Comment (beliayeu) [reply packages/gatsby-theme-bodiless/src/dist/GatsbyImage/asGatsbyImage.tsx]: fixed in ce9fb9b75a305e327cdf265fd377690f19c9191c

**Review by wodenx — APPROVED** [review]

**Comment by hvanyo** [comment]
@beliayeu  The alt is now being passed but the design isn't.   we can separate out as separate ticket if you wish..

Easiest way to test is is probably switch the test site to having the tout image serve a fluid img and then you will see styling with rounded corners is lost.

**Comment by hvanyo** [comment]
@beliayeu  not sure if its because I am not in a clean env but npm run build is throwing the error: @bodiless/test-site: /Users/heidivanyo/slate/bj-andrei/packages/gatsby-theme-bodiless/dist/fsLogHandler.js:52
@bodiless/test-site:         throw e;

**Review by hvanyo — APPROVED** [review]

**Pull request merged**